src/System.Net.Http/tests/FunctionalTests/CancellationTest.cs

```diff
@@ -2,161 +2,442 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
 using System.Diagnostics;
+using System.Linq;
 using System.IO;
 using System.Net.Test.Common;
 using System.Threading;
 using System.Threading.Tasks;
-
 using Xunit;
-using Xunit.Abstractions;
 
 namespace System.Net.Http.Functional.Tests
 {
-    public class CancellationTest : HttpClientTestBase
+    public class HttpClientHandler_CancellationTest : HttpClientTestBase
     {
-        private readonly ITestOutputHelper _output;
+        [OuterLoop]
```

Comment:
Since we're using LoopbackServer, should this be inner loop?

Reply:
> But historically, cancelation tests tend to be unreliable in general because they rely on timing.

These shouldn't be unreliable, as they use synchronization between client and server to order operations as desired.  There are some places where cancellation may occur here or there and thus we're not necessarily testing exactly the thing we hope to test, but the tests should still pass and be reliable from that perspective.

> Since we're using LoopbackServer, should this be inner loop?

I'll need to look again at how long the various tests take.  There are reasons besides loopback server to make things outerloop, e.g. they're not super fast, but IIRC all of these tests run within a second or two, so innerloop should be ok.